compiler/modules/front/codeGen.bal

```diff
@@ -363,11 +374,23 @@ function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Environment initialE
         else {
             env.assignments.push(...effect.assignments);
         }
+
+        StmtNarrowing[] narrowings = effect.narrowings;
+        //pr-todo: filter out narrowings that should't be visible to outside ?
+        env.narrowings.push(...narrowings);
+        if curBlock != () {
+            //pr-todo: narrowings at the end of a function is useless, can we prevent?
+            foreach var n in narrowings {
+                { bindings: bindings } = codeGenNarrowing(cx, curBlock, env, n.binding, n.completeNormally, n.basis, stmt.startPos);
+                env.bindings = bindings;
+            }
+        }
     }                
     check unusedLocalVariables(cx, env, initialEnv.bindings);
     int[] assignments = [];
     addAssignments(assignments, env.assignments, startRegister);
```

Comment:
I am not clear what your strategy is for dealing with assignments...

Reply:
Assignments worked even without me doing anything. Need to test a bit more to see if that works for all cases.